src/apps/Altinn.AccessManagement/src/Altinn.AccessMgmt.Persistence/Services/AssignmentService.cs

```diff
@@ -29,6 +30,136 @@ IDelegationRepository delegationRepository
     private readonly IRolePackageRepository rolePackageRepository = rolePackageRepository;
     private readonly IEntityRepository entityRepository = entityRepository;
     private readonly IDelegationRepository delegationRepository = delegationRepository;
+    private static readonly string RETTIGHETSHAVER = "rettighetshaver";
+    private static readonly Guid PartyTypeOrganizationUuid = new Guid("8c216e2f-afdd-4234-9ba2-691c727bb33d");
+
+    /// <inheritdoc/>
+    public async Task<IEnumerable<ClientDto>> GetClients(Guid toId, string[] roles, string[] packages, CancellationToken cancellationToken = default)
+    {
+        // Setup filter to be used for fetching client assignments
+        var clientFilter = assignmentRepository.CreateFilterBuilder();
+        clientFilter.Equal(t => t.ToId, toId);
+
+        // Fetch role metadata
+        var roleFilter = roleRepository.CreateFilterBuilder();
+        roleFilter.In(t => t.Code, roles);
+        var roleResult = await roleRepository.Get(roleFilter, cancellationToken: cancellationToken);
+
+        if (roleResult == null || !roleResult.Any() || roleResult.Count() != roles.Length)
+        {
+            throw new ArgumentException($"Filter: {nameof(roles)}, provided contains one or more role identifiers which cannot be found.");
+        }
+
+        var filterRoleIds = roleResult.Select(r => r.Id).ToList();
+
+        // Fetch role-package metadata
+        var rolePackFilter = rolePackageRepository.CreateFilterBuilder();
+        rolePackFilter.In(t => t.RoleId, filterRoleIds);
+        var rolePackageResult = await rolePackageRepository.Get(rolePackFilter, cancellationToken: cancellationToken);
+
+        // Fetch package metadata
+        var packageResult = await packageRepository.Get(cancellationToken: cancellationToken);
+
+        if (!packages.All(p => packageResult.Select(pr => pr.Urn).Contains($"urn:altinn:accesspackage:{p}")))
+        {
+            throw new ArgumentException($"Filter: {nameof(packages)}, provided contains one or more package identifiers which cannot be found.");
+        }
+
+        // Fetch client assignments
+        clientFilter.In(t => t.RoleId, filterRoleIds);
+        var clients = await assignmentRepository.GetExtended(clientFilter, cancellationToken: cancellationToken);
+
+        // Fetch assignment packages
+        QueryResponse<AssignmentPackage> assignmentPackageResult = null;
+        if (roles.Contains(RETTIGHETSHAVER))
+        {
+            var rettighetshaverClients = clients.Where(c => c.RoleId == roleResult.First(r => r.Code == RETTIGHETSHAVER).Id);
```

Comment:
her filtrerer vi bort kun rettighetshavere. på linje 96 så skipper vi alle parties som ikke er orgs basert på var assignments (som inneholder alle). Vil dette kunne bli et problem?

Reply:
Hm.. Poenget med å her filtrere til bare de assignmentene som gjelder Rettighetshavere er fordi det bare er de assignmentene som har AssignmentPackages knytt til seg. Så dette kallet er bare for å hente da informasjonen om hvilke pakker er delegert. 

I etterkant forkaster jeg alt som ikke er assignments fra organisasjoner ja. Det er hovedsakelig som en safe-guard til senere, når personer også kan begynne å delegere til organisasjoner. Da det fortsatt er litt usikkert om Person -> Org delegeringer også skal være en del av klientdelegeringsløsningen.
De andre rollene er jo bare for enhetsregisteret (pr. nå) så kunne ha flyttet det org-only filteret hit.. 🤔